test/unit/BeefyVaultOperator.unit.ts

```diff
@@ -0,0 +1,173 @@
+
+    before("loader", async () => {
+        loadFixture = createFixtureLoader(provider.getWallets(), provider);
+    });
+
+    beforeEach("create fixture loader", async () => {
+        context = await loadFixture(factoryAndOperatorsForkingBSCFixture);
+    });
+
+    it("deploys and has an address", async () => {
+        expect(context.beefyVaultOperator.address).to.be.a.string;
+    });
+
+    describe("deposit()", () => {
+        it("Should revert if amount to deposit is zero", async () => {
+            // All the amounts for this test
+            const bnbToDeposit = appendDecimals(1);
+            const bnbToDepositAndFees = bnbToDeposit.add(getExpectedFees(bnbToDeposit));
+
+            // Orders to deposit in beefy with amount 0
+            let orders: utils.OrderStruct[] = utils.getBeefyBnbVenusDepositOrder(context, BIG_NUMBER_ZERO);
+
+            // User1 creates the portfolio/NFT
+            await expect(
+                context.nestedFactory
+                    .connect(context.user1)
+                    .create(0, [{ inputToken: ETH, amount: bnbToDepositAndFees, orders, fromReserve: false }], {
+                        value: bnbToDepositAndFees,
+                    }),
+            ).to.be.revertedWith("NF: OPERATOR_CALL_FAILED");
+        });
+
+        it("Should revert if deposit more than available", async () => {
+            // All the amounts for this test
+            const bnbToDeposit = appendDecimals(1);
+            const bnbToDepositAndFees = bnbToDeposit.add(getExpectedFees(bnbToDeposit));
+
+            // Orders to deposit in beefy with "initial amount x 2" (more than msg.value)
+            let orders: utils.OrderStruct[] = utils.getBeefyBnbVenusDepositOrder(context, bnbToDepositAndFees.mul(2));
+
+            // User1 creates the portfolio/NFT
+            await expect(
+                context.nestedFactory
+                    .connect(context.user1)
+                    .create(0, [{ inputToken: ETH, amount: bnbToDepositAndFees, orders, fromReserve: false }], {
+                        value: bnbToDepositAndFees,
+                    }),
+            ).to.be.revertedWith("NF: OPERATOR_CALL_FAILED");
+        });
+
+        it("Create/Deposit in Beefy with BNB", async () => {
+            // All the amounts for this test
+            const bnbToDeposit = appendDecimals(1);
+            const bnbToDepositAndFees = bnbToDeposit.add(getExpectedFees(bnbToDeposit));
+
+            const ethBalanceBefore = await context.user1.getBalance();
+
+            // Orders to Deposit in beefy
+            let orders: utils.OrderStruct[] = utils.getBeefyBnbVenusDepositOrder(context, bnbToDeposit);
+
+            // User1 creates the portfolio/NFT
+            const tx = await context.nestedFactory
+                .connect(context.user1)
+                .create(0, [{ inputToken: ETH, amount: bnbToDepositAndFees, orders, fromReserve: false }], {
+                    value: bnbToDepositAndFees,
+                });
```

Comment:
🔧 I would like to also see a test case for deposits post-creation using the new process orders method

Reply:
👍